src/sdk/namespace_cache.js

```diff
@@ -4,19 +4,35 @@
 const _ = require('lodash');
 const stream = require('stream');
 const assert = require('assert');
+const dbg = require('../util/debug_module')(__filename);
 
 class NamespaceCache {
 
-    constructor({ namespace_hub, namespace_nb, active_triggers }) {
+    constructor({ namespace_hub, namespace_nb, caching, active_triggers }) {
         this.namespace_hub = namespace_hub;
         this.namespace_nb = namespace_nb;
         this.active_triggers = active_triggers;
+        this.caching = caching;
     }
 
     get_write_resource() {
         return this.namespace_hub;
     }
 
+    _delete_object_from_cache(params, object_sdk) {
+        process.nextTick(() => {
```

Comment:
Since cache delete does not need to block s3 procedure, client can get quicker response from the endpoint without waiting for cache delete to be finished. Because no await is used, by using nextTick, it allows the cache delete to be handled as soon as possible by node ( after the current operation in node is completed), so cache entry will be cleaned up before client's next s3 operation.

Reply:
First nextTick should probably be replaced with setImmediate because nextTick shouldn't be used unless no other option because it can starve IO events.
But in this case to submit an operation in the background, we usually write the async function as usual, so that if some callers want to await it they can do so, and for callers that do not want to wait we just call without awaiting, and the effect is the same. You could also do this on the call side to make it more visible:
```
setImmediate(() => this. _delete_object_from_cache(...));
```

Reply:
Ok. Will use try catch.